src/tests/unit/directory-dao.test.js

```diff
@@ -0,0 +1,74 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import _ from 'lodash';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+import * as directorySerializer from 'api/v2/serializers/directory-serializer';
+import { singleResult, multiResult } from './mock-data';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const proxyDao = (option) => {
+  const proxyquireObject = {
+    util: {},
+
+    './connection': {
+      getClient: sinon.stub().returns({ search: {} }),
+    },
+  };
+
+  if (option === 'getDirectories') {
+    proxyquireObject.util.promisify = () => sinon.stub().resolves({
+      on: (type, listener) => {
+        if (type === 'end') {
+          listener();
+        }
+      },
+    });
+  } else if (option === 'getDirectory') {
+    proxyquireObject.util.promisify = () => sinon.stub().resolves({
+      on: (type, listener) => {
+        if (type === 'searchEntry') {
+          listener({ object: {} });
+        }
+      },
+    });
+  }
+  return proxyquire('api/v2/db/ldap/directory-dao', proxyquireObject);
+};
+
+describe('Test directory-dao', () => {
+  afterEach(() => sinon.restore());
+
+  describe('Test getDirectory', () => {
+    const testCaseList = [singleResult];
+    _.forEach(testCaseList, ({ testCase, expectedResult, description }) => {
+      it(`getDirectory should be fulfilled with ${description}`, () => {
+        const serializeDirectoryStub = sinon.stub(directorySerializer, 'serializeDirectory');
+        serializeDirectoryStub.returns(testCase);
+        const result = proxyDao('getDirectory').getDirectory('fakeId');
+        return result.should.eventually.be.fulfilled.and.deep.equal(expectedResult);
+      });
+    });
+  });
+
+
+  describe('Test getDirectories', () => {
+    const testCaseList = [multiResult];
+    _.forEach(testCaseList, ({ testCase, expectedResult, description }) => {
+      it(`getDirectories should be fulfilled with ${description}`, () => {
+        const serializeDirectoriesStub = sinon.stub(directorySerializer, 'serializeDirectories');
+        serializeDirectoriesStub.returns(testCase);
+        const result = proxyDao('getDirectories').getDirectories({ firstName: 'fakeName' });
+        return result.should.eventually.be.fulfilled.and.deep.equal(expectedResult);
+      });
+    });
```

Comment:
These two cases are near identical. I would make a helper function that takes the differences as parameters. Having the serializer stub return `testCase` which is the same as `expectedResult` pretty much nullifies your tests. Try setting it to `returnsArg(0)`, I don't think that will break your tests. Then you can just always stub both serializers in the helper function and it won't need to be an additional parameter. Checkout the helper functions I wrote in [onbase unit tests](https://github.com/osu-mist/onbase-api/blob/master/src/tests/unit/onbase-dao.test.js) for inspiration.